pvlib/pvsystem.py

```diff
@@ -432,20 +452,54 @@ def sapm(self, effective_irradiance, temp_cell, **kwargs):
         """
         return sapm(effective_irradiance, temp_cell, self.module_parameters)
 
-    def sapm_celltemp(self, irrad, wind, temp):
-        """Uses :py:func:`sapm_celltemp` to calculate module and cell
-        temperatures based on ``self.racking_model`` and
-        the input parameters.
+    def sapm_celltemp(self, poa_global, temp_air, wind_speed,
+                      parameter_set=None):
+        """Uses :py:func:`temperature.sapm_cell` to calculate cell
+        temperatures.
 
         Parameters
         ----------
-        See pvsystem.sapm_celltemp for details
+        poa_global : numeric or Series
+            Total incident irradiance in W/m^2.
+
+        temp_air : numeric or Series
+            Ambient dry bulb temperature in degrees C.
+
+        wind_speed : numeric or Series
+            Wind speed in m/s at a height of 10 meters.
+
+        parameter_set : string, default None
+            Heat loss model parameters to be used.
+            See temperature.TEMPERATURE_MODEL_PARAMETERS for available
+            parameter sets.
 
         Returns
         -------
-        See pvsystem.sapm_celltemp for details
+        numeric or Series, values in degrees C.
         """
-        return sapm_celltemp(irrad, wind, temp, self.racking_model)
+        if parameter_set is not None:
+            kwargs = temperature._temperature_model_params('sapm',
+                                                           parameter_set)
+        else:
+            kwargs = _build_kwargs(['a', 'b', 'deltaT'],
+                                   self.temperature_model_parameters)
+        return temperature.sapm_cell(poa_global, temp_air, wind_speed,
+                                     **kwargs)
+
+    def _infer_temperature_model_params(self):
+        # try to infer temperature model parameters from from racking_model
+        # and module_type
+        param_set = self.racking_model + '_' + self.module_type
+        if param_set in temperature.TEMPERATURE_MODEL_PARAMETERS['sapm']:
+            return temperature._temperature_model_params('sapm', param_set)
+        elif 'freestanding' in param_set:
+            return temperature._temperature_model_params('pvsyst',
+                                                         'freestanding')
+        elif 'insulated' in param_set:  # after SAPM to avoid confusing keys
+            return temperature._temperature_model_params('pvsyst',
+                                                         'insulated')
+        else:
+            return {}
```

Comment:
Might be better to raise a `KeyError` here and `except KeyError: warnings.warn(deprecation)` in the `__init__`. But I am ok leaving as is.